src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -40,7 +40,7 @@
      */
     public function __construct($name, NodeInterface $parent = null)
     {
-        if (false !== strpos($name, '.')) {
+        if (null !== $name && false !== strpos($name, '.')) {
```

Comment:
why not cast to string instead?
`if (false !== strpos($name = (string) $name, '.')) {`

Reply:
yes, better. Now updated.

Reply:
Didnt reassign `$name` btw... we need to preserve string|null on the property side.

Reply:
> we need to preserve string|null on the property side

why? would need a side effect, which one is it? casting all the time is hurting perf, that's not the side effect I'd consider good ;)

Reply:
> which one is it?

BC... (`protected $name;`). Also the constructor sig kinda implies a nullable value.

Then again.. if i got this right, it shouldnt hurt indeed. AFAIK prototyped nodes will have a string name (`setName()`) once being dealt with. And the calling side needs to expect string anyway due `getName()`.

Given `$arr[null] = .. ` equals `$arr[''] = ...` (see remaining issue) i think it's safe to make the change :+1:

Reply:
> the constructor sig kinda implies a nullable value

not the docblock, isn't it? I'd support doing the change :)

Reply:
touche. It does in 4.0 though. Now updated.